packages/flutter/lib/src/rendering/list_wheel_viewport.dart

```diff
@@ -876,7 +905,7 @@ class RenderListWheelViewport
 
     final bool shouldApplyOffCenterDim = overAndUnderCenterOpacity < 1;
     if (useMagnifier || shouldApplyOffCenterDim) {
-      _paintChildWithMagnifier(context, offset, child, transform, offsetToCenter, untransformedPaintingCoordinates);
+      _paintChildWithMagnifier(context, offset, child, transform, offsetToCenter, untransformedPaintingCoordinates, center: center);
     } else {
       _paintChildCylindrically(context, offset, child, transform, offsetToCenter);
```

Comment:
should we assert that center is null if we go down the else branch?

Actually, the fact that this branch ignores center silently makes the code kinda hard to follow (and I think it is also the source of a bug in the current implementation, see my comment about useMagnifier above). Maybe we need to refactor this somehow to make it easier to follow.

Reply:
> should we assert that center is null if we go down the else branch?

What about this one? Since this is already pretty confusing it may be worthwhile to put a guardrail in place for when this gets refactored to make the caller instantly aware of the fact that their specification of center is not doing what they may think it is doing?

Reply:
I added the assert. I'm not in love with this code but I don't see a nice way to re-arrange it all...